crates/contract/src/call.rs

```diff
@@ -368,30 +383,25 @@ impl<P: TempProvider, D: CallDecoder> CallBuilder<P, D> {
     /// Note that the deployment address can be pre-calculated if the `from` address and `nonce` are
     /// known using [`calculate_create_address`](Self::calculate_create_address).
     pub async fn deploy(&self) -> Result<Address> {
-        if self.request.to.is_some() {
+        if !self.request.to().is_some_and(|to| to.is_create()) {
             return Err(Error::NotADeploymentTransaction);
         }
         let pending_tx = self.send().await?;
         let receipt = pending_tx.await?;
-        receipt.contract_address.ok_or(Error::ContractNotDeployed)
+        receipt
+            .ok_or(Error::ContractNotDeployed)?
+            .contract_address()
+            .ok_or(Error::ContractNotDeployed)
     }
 
     /// Broadcasts the underlying transaction to the network.
     // TODO: more docs referring to customizing PendingTransaction
-    pub async fn send(&self) -> Result<impl IntoFuture<Output = Result<TransactionReceipt>>> {
-        // TODO: send_transaction, PendingTransaction
-        // NOTE: This struct is needed to have a concrete type for the `Future` trait.
-        struct Tmp<T>(PhantomData<T>);
-        impl<T> Future for Tmp<T> {
-            type Output = T;
-            fn poll(
-                self: Pin<&mut Self>,
-                _cx: &mut std::task::Context<'_>,
-            ) -> std::task::Poll<Self::Output> {
-                todo!()
-            }
-        }
-        Ok(Tmp(PhantomData))
+    pub async fn send(
+        &self,
+    ) -> Result<impl IntoFuture<Output = Result<Option<N::ReceiptResponse>>> + '_> {
+        let pending = self.provider.send_transaction(self.request.clone()).await?;
```

Comment:
todo in followup, return pending transaction

Reply:
+1 i think not a blocker for the foundry integration given we await the receipt in most areas w anvil

Reply:
https://github.com/alloy-rs/alloy/issues/264